### ISSUE TITLE
Alamofire building unspecify scheme  platformwhen using Carthage even using --platform parameters

### ISSUE THREAD

**Issue by vonghiep** [Alamofire/Alamofire#3889]
> ℹ Please fill out this template when filing an issue.
> All lines beginning with an ℹ symbol instruct you with what info we expect. You can delete those lines once you've filled in the info.
>
> Per our [*CONTRIBUTING guidelines](https://github.com/Alamofire/Alamofire/blob/master/CONTRIBUTING.md), we use GitHub for
> bugs and feature requests, not general support. Other issues should be opened on Stack Overflow with the tag `alamofire`.
>
> Please remove this line and everything above it before submitting.

* [x] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/Alamofire/Alamofire/blob/master/CONTRIBUTING.md).

## What did you do?

I updated my Xcode to 16 Beta to test on iOS 18 and handle bugs, new features, etc..

I had Alamofire in my Cart file. Just plant simple setting 
```
github "Alamofire/Alamofire" == 5.9.1
```
And then I run 
```
carthage bootstrap Alamofire --platform iOS--use-xcframeworks --use-ssh --no-use-binaries --verbose
```
to update Alamofire

**More Context Why `carthage bootstrap`**: My app is a pretty old app it using Moya and ReactiveSwift, which is a hell junk of code because it "updated" ReactiveSwift to 7.1.1 which no longer compatible with Moya 15.0.3 and we did get slideby by using `carthage bootstrap` instead of `carthage build` - On my way to delete Moya but I don't think it can catch up with iOS 18 release unless I invest all time in it

## What did you expect to happen?

After I run this command
```
carthage bootstrap Alamofire --platform iOS--use-xcframeworks --use-ssh --no-use-binaries --verbose
```
It rebuild Alamofire with new Command Line Tools 16.1 from Xcode 16

## What happened instead?

it keep showing the error log
```
A shell task (/usr/bin/xcrun xcodebuild -project /Users/xxx/Carthage/Checkouts/Alamofire/watchOS\ Example/watchOS\ Example.xcodeproj -scheme watchOS\ Example\ WatchKit\ App -configuration Release CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -showBuildSettings -skipUnavailableActions) failed with exit code 6:
2024-08-15 14:43:37.415 xcodebuild[31337:254751] Writing error result bundle to /var/folders/z6/qj82nz8d5zs6lyvn8l_dw4wc0000gq/T/ResultBundle_2024-15-08_14-43-0037.xcresult
xcodebuild: error: Found no destinations for the scheme 'watchOS Example WatchKit App' and action archive.
```
or sometime
```
A shell task (/usr/bin/xcrun xcodebuild -project /Users/xxx/Carthage/Checkouts/Alamofire/Alamofire.xcodeproj -scheme Alamofire\ tvOS -configuration Release CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -showBuildSettings -skipUnavailableActions) failed with exit code 6:
2024-08-15 14:43:04.902 xcodebuild[31199:253239] Writing error result bundle to /var/folders/z6/qj82nz8d5zs6lyvn8l_dw4wc0000gq/T/ResultBundle_2024-15-08_14-43-0004.xcresult
xcodebuild: error: Found no destinations for the scheme 'Alamofire tvOS' and action archive.
```
or sometime even visionOS one.
The problem is why building scheme tvOS or watchOS when I did specify the `--platform iOS` parameter in the cmd line

And this only happen in XCode 16 Beta, everything fine with Xcode 15.3

## Alamofire Environment

**Alamofire Version:** 5.9.1
**Dependency Manager:** Carthage
**Xcode Version:** Xcode 16.1
**Swift Version:** Swift 5
**Platform(s) Running Alamofire:** iOS
**macOS Version Running Xcode:** Sonoma 14.6.1

**Comment by HoonHaChoi** [comment]
Try running **brew uninstall carthage; brew install --head carthage** 
This is how I was able to solve the issue.

[https://github.com/Carthage/Carthage/issues/3375](https://github.com/Carthage/Carthage/issues/3375)

Alamofire Version: 5.9.1
Dependency Manager: Carthage
Xcode Version: Xcode 16.0 RC
Swift Version: Swift 5

**Comment by jshier** [comment]
This appears to be a Carthage issue, not Alamofire, so I suggest you check your Carthage version (they just released a 0.40.0 update) and try again. Alamofire really has little effect on how Carthage integrates it, as Carthage just uses our project file and builds the targets we have.
